app/src/org/commcare/android/util/TemplatePrinterUtils.java

```diff
@@ -0,0 +1,184 @@
+        BufferedReader in = new BufferedReader(new FileReader(file));
+        String str;
+        while ((str = in.readLine()) != null) {
+            builder.append(str);
+        }
+        in.close();
+        return builder.toString();
+    }
+
+    /**
+     * Writes the given string, encrypted, to the file location specified
+     *
+     * @param fileText the content to write
+     * @param outputPath the path of the file to write to
+     * @throws IOException
+     */
+    public static void writeStringToFile(String fileText, String outputPath) throws IOException {
+        try {
+            Cipher encrypter = Cipher.getInstance("AES");
+            encrypter.init(Cipher.ENCRYPT_MODE, KEY);
+            FileOutputStream fos = new FileOutputStream(new File(outputPath));
+            CipherOutputStream cos = new CipherOutputStream(fos, encrypter);
+            BufferedWriter out = new BufferedWriter(new OutputStreamWriter(cos));
+            out.write(fileText);
+            out.close();
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        } catch (NoSuchPaddingException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        } catch (InvalidKeyException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Reads back from the encrypted file generated by the above and returns a string
+     * representation of the file's contents
+     *
+     * @param readFromPath the path to the file from which to read
+     * @return a string representation of the file's contents
+     */
+    public static String readStringFromFile(String readFromPath) throws IOException {
+        try {
+            Cipher decrypter = Cipher.getInstance("AES");
+            decrypter.init(Cipher.DECRYPT_MODE, KEY);
+            FileInputStream fis = new FileInputStream(new File(readFromPath));
+            CipherInputStream cis = new CipherInputStream(fis, decrypter);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(cis));
+            String docString = "";
+            String line;
+            while ((line = reader.readLine()) != null) {
+                docString += line;
+            }
+            reader.close();
+            return docString;
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        } catch (NoSuchPaddingException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        } catch (InvalidKeyException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
```

Comment:
Since all of these catches have the same body, you can squash them by using new java 7 syntax:
```
catch (Exception1 | Exception2 | ... | ExceptionN e) { ...
```